packages/algob/test/internal/core/config/config-validation.ts

```diff
@@ -608,4 +608,148 @@ describe("Config validation", function () {
+          }
+        ]
+      };
+      expectBuilderError(
+        () =>
+          validateConfig({
+            networks: {
+              localhost: Object.assign(localhost, kmdCfg),
+              [ALGOB_CHAIN_NAME]: {
+                asdasd: "1"
+              }
+            }
+          }),
+        ERRORS.GENERAL.INVALID_CONFIG
+      );
+    });
+
+    it("Shouldn't accept invalid Account name", function () {
+      const kmdCfg = {
+        host: "127.0.0.1",
+        port: 8080,
+        token: "some_kmd_token",
+        wallets: [
+          {
+            name: "Wallet",
+            password: "",
+            accounts: [
+              { name: 98712, address: "addr-2" }]
+          }
+        ]
+      };
+      expectBuilderError(
+        () =>
+          validateConfig({
+            networks: {
+              localhost: Object.assign(localhost, kmdCfg),
+              [ALGOB_CHAIN_NAME]: {
+                asdasd: "2"
+              }
+            }
+          }),
+        ERRORS.GENERAL.INVALID_CONFIG
+      );
+    });
+
+    it("Shouldn't accept invalid address", function () {
+      const kmdCfg = {
+        host: "127.0.0.1",
+        port: 8080,
+        token: "some_kmd_token",
+        wallets: [
+          {
+            name: "Wallet",
+            password: "",
+            accounts: [
+              { name: "Account1", address: ["addr-1"] }]
+          }
+        ]
+      };
+      expectBuilderError(
+        () =>
+          validateConfig({
+            networks: {
+              localhost: Object.assign(localhost, kmdCfg),
+              [ALGOB_CHAIN_NAME]: {
+                asdasd: "3"
+              }
+            }
+          }),
+        ERRORS.GENERAL.INVALID_CONFIG
+      );
+    });
+
+    it("Shouldn't fail with valid KMD config", function () {
+      const errors = getValidationErrors({
+        networks: {
+          localhost: localhost.kmdCfg = {
+            host: "127.0.0.1",
+            port: 8080,
+            token: "some_kmd_token",
+            wallets: [
+              {
+                name: "Wallet",
+                password: "",
+                accounts: [
+                  { name: "Account1", address: "addr-3" }]
+              }
+            ]
+          },
+          [ALGOB_CHAIN_NAME]: {
+            asdasd: "1234"
+          }
+        }
+      });
+
+      assert.isEmpty(errors.errors, errors.toString());
+    });
+
+    it("Shouldn't fail with unrecognized params", function () {
+      const errors = getValidationErrors({
+        networks: {
+          localhost: localhost.kmdCfg = {
+            host: "127.0.0.1",
+            port: 8080,
+            token: "some_kmd_token",
+            wallets: [
+              {
+                name: "Wallet",
+                password: "",
+                accounts: [
+                  { name: "Account1", address: "addr-4" }]
+              }
+            ],
+            otherParam: "some_other_detail"
+          },
```

Comment:
you can store a valid config in variable and reuse it in 2 tests instead of copying lot of lines.
Here, in this test you can do:
```
cfg.networks.localhost.kmdCfg.otherParam = "some_other_detail"
```

Reply:
In other tests you can copy the object using `deepmerge` (we already have it in dependencies):
```
let invalidCfg = deepmerge({}, cfg)
```

Reply:
ok

Reply:
![asd](https://user-images.githubusercontent.com/26249903/98079318-93986600-1e99-11eb-89dd-6598e0b8a0bc.png)

If I do it using `deepmerge`, it doesn't let me assign wrong types to values, but user can put wrong types in `algob.config.js`

Reply:
Ah, so maybe using a function will be a good solution?

```
let cfg = mkCfg()
```

Reply:
![work](https://user-images.githubusercontent.com/26249903/98096655-06154000-1eb2-11eb-90db-ed992fe81ecc.png)

this works if i use any in tests.

Reply:
![works](https://user-images.githubusercontent.com/26249903/98096879-3f4db000-1eb2-11eb-8be2-24b2252d670a.png)

this works if i use object.assign()

Reply:
![work](https://user-images.githubusercontent.com/26249903/98096979-5e4c4200-1eb2-11eb-9bee-c559182967db.png)

this works if i use `any` with `object.assign()`

Reply:
![work1](https://user-images.githubusercontent.com/26249903/98101463-411a7200-1eb8-11eb-8356-d067aa8089e3.png)

works when i use `any`